pkg/api/types.go

```diff
@@ -30,36 +30,25 @@ type ContainerService struct {
 
 // Properties represents the ACS cluster definition
 type Properties struct {
-	ProvisioningState   ProvisioningState    `json:"provisioningState,omitempty"`
-	OrchestratorProfile *OrchestratorProfile `json:"orchestratorProfile,omitempty"`
-	MasterProfile       *MasterProfile       `json:"masterProfile,omitempty"`
-	AgentPoolProfiles   []*AgentPoolProfile  `json:"agentPoolProfiles,omitempty"`
+	ProvisioningState       ProvisioningState        `json:"provisioningState,omitempty"`
+	OrchestratorProfile     *OrchestratorProfile     `json:"orchestratorProfile,omitempty"`
+	MasterProfile           *MasterProfile           `json:"masterProfile,omitempty"`
+	AgentPoolProfiles       []*AgentPoolProfile      `json:"agentPoolProfiles,omitempty"`
 	ServicePrincipalProfile *ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
-	AzProfile *AzProfile `json:"azProfile,omitempty"`
+	AzProfile               *AzProfile               `json:"azProfile,omitempty"`
 }
 
 // AzProfile holds the azure context for where the cluster resides
 type AzProfile struct {
```

Comment:
I thought we were getting rid of these options (they wont be provided by end users) and the RP would enrich the config with them?

Reply:
still used on the cmd type to simulate enrichment so we need them on the internal struct

Reply:
https://github.com/pweil-/openshift-azure/blob/d438b41ea1a2df590f208dfc388ee62352808a84/cmd/createorupdate/createorupdate.go#L106

Reply:
Hrm, so if these are our internal types, when do we plan to add the external? Can at least have a comment here to mention that this is only used internally and should be removed from external once we copy.

Reply:
external is in v1.  We need the RP to enrich the internal type for us and save it.  I'm not sure we need it on the external type.  WDYT?  I was thinking that if we state the RP will *always* enrich those fields then we don't need to worry about stability in conversion and if we (SRE) are always looking at the saved type then the values should always be there

Reply:
https://github.com/openshift/openshift-azure/pull/143/commits/1707b273cd9d0172bf0e3c49a6a9bb04df23784e for quick reference to v1 :smile:

Reply:
cool, not sure how I missed that. We don't need it in the external type.